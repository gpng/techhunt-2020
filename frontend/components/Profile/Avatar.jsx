import React from 'react';
import PropTypes from 'prop-types';
// components
import { COLORS } from '../../constants/styles';

const Avatar = ({ width }) => {
  return (
    <div className="avatar-root">
      <style jsx>{`
        .avatar-root {
          width: ${width};
          height: ${width};
          border-radius: 50%;
          background: ${COLORS.BACKGROUND_PRIMARY};
          border: 2px solid ${COLORS.BACKGROUND_DARK_SECONDARY};
        }
      `}</style>
    </div>
  );
};

Avatar.defaultProps = {
  width: '6rem',
};

Avatar.propTypes = {
  width: PropTypes.string,
};

export default Avatar;
