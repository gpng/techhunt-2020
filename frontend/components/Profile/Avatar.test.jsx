import { render, cleanup } from '@testing-library/react';
import Avatar from './Avatar';

afterEach(cleanup);

it('should render', () => {
  render(<Avatar />);
});

it('should match default snapshot', () => {
  const { asFragment } = render(<Avatar />);
  expect(asFragment(<Avatar />)).toMatchSnapshot();
});

it('should match snapshot with width', () => {
  const props = {
    width: '10rem',
  };
  const { asFragment } = render(<Avatar width={props.width} />);
  expect(asFragment()).toMatchSnapshot();
});
