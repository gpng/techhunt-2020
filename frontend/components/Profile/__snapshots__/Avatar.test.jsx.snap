// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match default snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-961371421 avatar-root"
  >
    <img
      alt=""
      class="jsx-961371421"
      src="/static/images/avatar_default.svg"
    />
  </div>
</DocumentFragment>
`;

exports[`should match snapshot with width 1`] = `
<DocumentFragment>
  <div
    class="jsx-1161802909 avatar-root"
  >
    <img
      alt=""
      class="jsx-1161802909"
      src="/static/images/avatar_default.svg"
    />
  </div>
</DocumentFragment>
`;
